samples/tfx/taxi-cab-classification-pipeline.py

```diff
@@ -14,128 +14,22 @@
-    return dsl.ContainerOp(
-        name = step_name,
-        image = 'gcr.io/ml-pipeline/ml-pipeline-dataflow-tfma:2c2445df83fa879387a200747cc20f72a7ee9727',
-        arguments = [
-            '--model', model,
-            '--eval', evaluation_data,
-            '--schema', schema,
-            '--project', project,
-            '--mode', analyze_mode,
-            '--slice-columns', analyze_slice_column,
-            '--output', '%s/{{workflow.name}}/analysis' % analysis_output,
-        ],
-        file_outputs = {'analysis': '/output.txt'}
-    )
-
-
-def dataflow_tf_predict_op(evaluation_data: 'GcsUri', schema: 'GcsUri[text/json]', target: str, model: 'TensorFlow model', predict_mode, project: 'GcpProject', prediction_output: 'GcsUri', step_name='prediction'):
-    return dsl.ContainerOp(
-        name = step_name,
-        image = 'gcr.io/ml-pipeline/ml-pipeline-dataflow-tf-predict:a277f87ea1d4707bf860d080d06639b7caf9a1cf',
-        arguments = [
-            '--data', evaluation_data,
-            '--schema', schema,
-            '--target', target,
-            '--model',  model,
-            '--mode', predict_mode,
-            '--project', project,
-            '--output', '%s/{{workflow.name}}/predict' % prediction_output,
-        ],
-        file_outputs = {'prediction': '/output.txt'}
-    )
-
-
-def confusion_matrix_op(predictions: 'GcsUri', output: 'GcsUri', step_name='confusion_matrix'):
-  return dsl.ContainerOp(
-      name=step_name,
-      image='gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:a277f87ea1d4707bf860d080d06639b7caf9a1cf',
-      arguments=[
-        '--output', '%s/{{workflow.name}}/confusionmatrix' % output,
-        '--predictions', predictions,
-        '--target_lambda', """lambda x: (x['target'] > x['fare'] * 0.2)""",
-     ])
-
-
-def roc_op(predictions: 'GcsUri', output: 'GcsUri', step_name='roc'):
-  return dsl.ContainerOp(
-      name=step_name,
-      image='gcr.io/ml-pipeline/ml-pipeline-local-roc:a277f87ea1d4707bf860d080d06639b7caf9a1cf',
-      arguments=[
-        '--output', '%s/{{workflow.name}}/roc' % output,
-        '--predictions', predictions,
-        '--target_lambda', """lambda x: 1 if (x['target'] > x['fare'] * 0.2) else 0""",
-     ])
-
-
-def kubeflow_deploy_op(model: 'TensorFlow model', tf_server_name, step_name='deploy'):
-    return dsl.ContainerOp(
-        name = step_name,
-        image = 'gcr.io/ml-pipeline/ml-pipeline-kubeflow-deployer:a277f87ea1d4707bf860d080d06639b7caf9a1cf',
-        arguments = [
-            '--model-export-path', '%s/export/export' % model,
-            '--server-name', tf_server_name
-        ]
-    )
+import kfp
+from kfp import components
+from kfp import dsl
+from kfp import gcp
+
+
+dataflow_tf_data_validation_op  = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/deb4d0ed41662055031832f4ba03e324dd609143/components/dataflow/tfdv/component.yaml')
```

Comment:
i thought we are going to deprecate the intermediate yaml?
@hongye-sun @gaoning777 what is the plan?

Reply:
The fate of intermediate yaml (the data format passed to the backend) is currently unclear, but the re-usable components are here to stay.
Loading declarative component definition files provides better experience for the users compared to copy-pasting the code snippets between pipelines and notebooks and makes the pipeline code files much shorter. (Of course nobody prevents the user from doing things the old copy-paste way.)

The recently released AI-Hub showcases that experience - component authors publish components and the pipeline authors can just load and start using them. We now have more than 50 components (AI-Hub and this repo) and that number is growing. @hongye-sun is a big contributor to this success.